src/catkin_pkg/package.py

```diff
@@ -226,10 +229,16 @@ def validate(self):
         if self.is_metapackage():
             if not self.has_buildtool_depend_on_catkin():
                 # TODO escalate to error in the future, or use metapackage.validate_metapackage
-                print('WARNING: Metapackage "%s" must buildtool_depend on catkin.' % self.name, file=sys.stderr)
+                new_warnings.append('Metapackage "%s" must buildtool_depend on catkin.' % self.name)
             if self.has_invalid_metapackage_dependencies():
-                print(('WARNING: Metapackage "%s" should not have other dependencies besides a ' +
-                       'buildtool_depend on catkin and run_depends.') % self.name, file=sys.stderr)
+                new_warnings.append('Metapackage "%s" should not have other dependencies besides a '
+                                    'buildtool_depend on catkin and run_depends.' % self.name)
+
+        for warning in new_warnings:
+            if warnings is None:
+                print('WARNING: ' + warning, file=sys.stderr)
+            elif warning not in warnings:
+                warnings += warning
```

Comment:
If `warnings` is a list and `warning` is a string than this operation won't do the right thing?